src/Server.cpp

```diff
@@ -656,6 +626,12 @@ void cServer::KickUser(int a_ClientID, const AString & a_Reason)
 
 void cServer::AuthenticateUser(int a_ClientID, const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
+	if (GetNumPlayers() >= GetMaxPlayers())
```

Comment:
Indeed, I may have tried something similar in d233e9843148313c71fbaba96ccff660e47b07b1. However, would this method have the potential to cause two clients vying for one remaining slot to both get kicked, with no one being let in?